### PR TITLE
feat(devices): sort devices alphabetically by name

### LINKED ISSUE #324: feat: sort device list alphabetically for better UX
## Problem

Currently, the device list is sorted by last_seen DESC (backend) which creates an unstable order — devices jump around as discovery pings update last_seen. This makes it hard for users to quickly find a specific device, especially with multiple devices.

## Proposed Solution

Sort devices **alphabetically by name** (case-insensitive) both in:
1. Backend API response (/api/devices)
2. Frontend display (DeviceSwiper, all device lists)

## Benefits

- **Stable order** — devices stay in the same position
- **Predictable** — users know where to look
- **Faster navigation** — especially with 3+ devices

### Backend
apps/backend/src/opencloudtouch/devices/repository.py:
ORDER BY LOWER(name) ASC

### Frontend
Already uses backend order — no changes needed if backend sorts correctly.

## Alternative Considered

- Sort by id (insertion order) — stable but not intuitive
- Keep last_seen — rejected due to instability

## Acceptance Criteria

- [x] Backend returns devices sorted A-Z by name (case-insensitive)
- [x] DeviceSwiper shows devices in alphabetical order
- [x] Order remains stable across page reloads/discovery pings

### PR DESCRIPTION
**Pull request by songdenkbar**
## Summary

Implements #324 by sorting devices alphabetically by name in the backend API using Unicode-aware, case-insensitive sorting.

## Changes

* Replaced `last_seen DESC` ordering with Python-based alphabetical sorting by device name
* Added Unicode normalization and case folding for names with umlauts
* Added a regression test covering case-insensitive sorting and umlauts
* Documented the alphabetical sort order in `openapi.yaml`

## Behavior Changes

Device ordering changes from `last_seen DESC` to alphabetical sorting by name.

Newly discovered devices will no longer automatically appear at the top of the list. Instead, they will appear at their alphabetical position, providing a stable and predictable device order.

Sorting is case-insensitive and handles umlauts consistently, e.g. `alpha`, `Ärger`, `Bravo`, `Zulu`.

## Testing

* `tests/unit/devices/test_repository.py`
* 27/27 tests passed

Closes #324

### CODE REVIEW

**Comment by scheilch** [comment]
Thank you for tackling #324! Alphabetical sorting is definitely the right approach here — having devices jump around based on `last_seen` is confusing, and stable ordering improves UX significantly.

The core idea and implementation direction are solid, but there are **two critical issues** that need fixing before we can merge this:

---

## Must fix before merge

### 🔴 CRITICAL: SQLite's `LOWER()` doesn't handle umlauts correctly

SQLite's `LOWER()` function only converts ASCII characters (A-Z). It **does not lowercase** international characters like Ä/Ö/Ü. This causes incorrect sorting — for example:

- **Expected:** ["alpha", "Ärger", "Bravo"]
- **Actual result:** ["alpha", "Bravo", "Ärger"]

This happens because the byte value of 'Ä' falls after 'Z' in the default collation.

**Fix:** Use SQLite's `COLLATE NOCASE` instead, which is both case-insensitive **and** Unicode-aware:

```sql
ORDER BY name COLLATE NOCASE ASC
```

This has been supported since SQLite 3.7.16+ and is the idiomatic way to do case-insensitive sorting in SQLite.

**Why this matters for OpenCloudTouch:** The codebase already uses German names with umlauts in tests (e.g., "Büro"), so this is a real-world issue for this project.

---

### 🔴 CRITICAL: Test gives false positive (only ASCII names tested)

The new test `test_device_get_all_sorted_by_name_case_insensitive` only uses ASCII names:

```python
"Zulu", "alpha", "Bravo"
```

This means the umlaut bug described above **goes undetected** — the test passes even though the feature is broken for international characters.

**Suggestion:** Add a test case with an umlaut to verify correct Unicode sorting:

```python
# Expected order: "alpha", "Ärger", "Bravo", "Zulu"
devices = [
    ("device1", "Zulu", ...),
    ("device2", "alpha", ...),
    ("device3", "Bravo", ...),
    ("device4", "Ärger", ...),  # Must sort before "Bravo"
]
```

After applying the `COLLATE NOCASE` fix, this test should verify that "Ärger" correctly appears between "alpha" and "Bravo".

---

### 🟡 HIGH: Sorting behavior not documented in OpenAPI spec

The `/api/devices` GET endpoint in `openapi.yaml` currently has an empty `schema: {}` and doesn't mention sort order.

**Suggestion:** Add a note in the endpoint description that devices are returned "sorted alphabetically by name (case-insensitive)". This helps contributers understand the behavior.

---

### 🟡 HIGH: Breaking change (sort order) should be noted

**Before this PR:** `ORDER BY last_seen DESC` → newly discovered devices appeared at the top  
**After this PR:** Alphabetical → new devices can appear anywhere in the list (including at the bottom)

This is an **acceptable trade-off** (stability > discovery convenience), but it's a behavioral change that should be mentioned in:
- The PR description (under a "Breaking Changes" or "Behavior Changes" section)

This helps users understand why the list order changed after an upgrade.

---

## Nice to have (optional)

**Edge case:** Device names starting with special characters (e.g., `@Kitchen`, `#Living`) will sort **before** alphabetical names (per ASCII rules). This is technically correct, but currently untested/undocumented. Consider adding a test case if this is relevant to your use case.

---

Once all these issues are addressed,  this will be ready to merge.

**Comment by songdenkbar** [comment]
Hi @scheilch , thanks for the review! One small concern: SQLite’s built-in `NOCASE` collation appears to be ASCII-only.

SQLite documentation:

> “only ASCII characters are case folded”
> https://www.sqlite.org/datatype3.html

SQLite FAQ:

> “Case-insensitive matching of Unicode characters does not work.”
> https://www.sqlite.org/faq.html#q18

So I’m wondering whether `COLLATE NOCASE` would still fail for umlauts such as `Ä`. Would Python-side Unicode normalization/casefolding be safer here — or am I missing something?

**Comment by scheilch** [comment]
You're right, sorry for the bad info — I tested it and `COLLATE NOCASE` behaves exactly like `LOWER(name)` here, both leave umlauts unfolded (`['alpha', 'Bravo', 'Zulu', 'Ärger']` in both cases). It's ASCII-only, as you found.

SQLite has no Unicode-aware collation without ICU, so let's sort in Python instead:

```python
devices = [self._row_to_device(row) for row in rows]
return sorted(devices, key=lambda d: unicodedata.normalize("NFKD", d.name).casefold())
```

Gives `["alpha", "Ärger", "Bravo", "Zulu"]`. Please add an umlaut test case once that's in.

File: apps/backend/src/opencloudtouch/devices/repository.py [hunk 1]
```diff
@@ -207,10 +208,14 @@ async def get_all(self) -> List[Device]:
...
-            ORDER BY last_seen DESC
         """)
 
         rows = await cursor.fetchall()
+        devices = [self._row_to_device(row) for row in rows]
+        return sorted(
+            devices,
+            key=lambda d: unicodedata.normalize("NFKD", d.name).casefold(),
+        )
         return [self._row_to_device(row) for row in rows]
```
Comment (scheilch) [inline apps/backend/src/opencloudtouch/devices/repository.py]: This should be deleted.

**Review by scheilch — CHANGES_REQUESTED** [review]

**Comment by songdenkbar** [comment]
All clear, done. 🙂

All core CI checks pass, but the Post CI summary as PR comment step fails with:

Unhandled error: HttpError: Resource not accessible by integration - https://docs.github.com/rest/issues/comments#create-an-issue-comment

Please let me know if there’s anything I should do on my side.

**Review by scheilch — APPROVED** [review]
